Editor/EditorCore/UndoUtility.cs

```diff
@@ -44,6 +44,25 @@ static void UndoRedoPerformed()
 #endif
         }
 
+        static int s_PreviewGroupIndex = -1;
+
+        internal static void StartPreview()
+        {
+            s_PreviewGroupIndex = Undo.GetCurrentGroup();
+        }
+
+        internal static void EndPreview()
+        {
+            if (s_PreviewGroupIndex != -1)
+            {
+                //Using this Undo method to remove the preview actions from the redo stack
+                Undo.RevertAllDownToGroup(s_PreviewGroupIndex);
+                s_PreviewGroupIndex = -1;
+            }
+            else
+                Undo.PerformUndo();
```

Comment:
this would be an excellent candidate for an `IDisposable` scope

Reply:
Done in [b7b9249](https://github.com/Unity-Technologies/com.unity.probuilder/pull/488/commits/b7b924932d392a26337da3e8804abf653f3655cd)